crates/pathfinder/src/storage/state.rs

```diff
@@ -567,6 +567,38 @@ impl StarknetTransactionsTable {
         Ok(data)
     }
 
+    pub fn get_transactions_for_latest_block(
+        sqlite_tx: &Transaction<'_>,
+    ) -> anyhow::Result<Vec<transaction::Transaction>> {
+        let block_hash = match StarknetBlocksTable::get(sqlite_tx, StarknetBlocksBlockId::Latest)? {
+            Some(block) => block.hash,
+            None => return Ok(Vec::new()),
+        };
+
+        let mut stmt = sqlite_tx
+            .prepare("SELECT tx FROM starknet_transactions WHERE block_hash = ? ORDER BY idx ASC")
+            .context("Preparing statement")?;
+
+        let mut rows = stmt
+            .query(params![block_hash.0.as_be_bytes()])
```

Comment:
Does this not work as:

```suggestion
            .query([block_hash.0.as_be_bytes()])
```

We should start adding those ToSql impls on starkhash wrappers to get to:


```
            .query([&block_hash])
```

But of course with the single query this would become parameterless.